services/send_user_report.py

```diff
@@ -0,0 +1,63 @@
+import webapp2
+import json
+import datetime
+import logging
+
+from google.appengine.api import urlfetch
+from google.appengine.api import mail
+from google.appengine.api import app_identity
+
+from models.connexus_user import ConnexusUser
+
+class SendUserReport(webapp2.RequestHandler):
+
+    def get(self):
```

Comment:
I think a post method makes more sense here, since the response is just an ok.

Reply:
I agree, however -- this is the method called by the cron job, which may only use HTTP 
get by definition. I am using the "rate" param (api/send_user_report?rate={1,2,3}) for the 3 separate types of reports. I'm sure a cleaner solution for a production app would be to store the last time you sent a report and to run the cron every minute, and compare every single user's rate to their last sent timestamp. I just went with the easiest for now. Do you think this should be updated?

Reply:
Yeah, the only alternative I can think of is to create more /api/ methods, instead of passing parameters, all of these implementations would eventually call a core function that receives the update_rate.
So the cron.yaml would end up something like:
```
- description: User Reporting 5 minutes
  url: /api/send_user_five_minutes
  schedule: every 5 minutes
- description: User Reporting 1 hour
  url: /api/send_user_hourly
  schedule: every 1 hours
- description: User Reporting 1 day
  url: /api/send_user_daily
  schedule: every 24 hours
```
That would mean adding more routes to the miniproject.py.
I think there should be an easier way, but what you are proposing is fine for now. Let's just have it in mind.

Reply:
This was my first approach and then just realized it was just code duplication and it felt.. dirtier? haha. I'll keep what we have for now but it's also definitely a lazy implementation. I don't think we need this method for the app version either since the cron is running from the cloud and is independent. We'll re-evaluate if needed :)